test/include/seqan3/test/tmp_directory.hpp

```diff
@@ -0,0 +1,205 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2021-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2021-2021, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#pragma once
+
+/*
+ * Internal test infrastructure.
+ *
+ * Define some helper classes and functions for the tests, that would be misplaced in the seqan3/include directory.
+ * @author Simon Gene Gottlieb <simon.gottlieb AT fu-berlin.de>
+ */
+
+#if defined(__APPLE__)
+#include <unistd.h>
+#elif defined(_WIN32)
+#include <cstring>
+#include <io.h>
+#else  // other unix systems
+#include <cstdlib>
+#endif
+
+#include <seqan3/std/filesystem>
+#include <iostream>
+
+#include <seqan3/core/platform.hpp>
+#include <seqan3/test/sandboxed_path.hpp>
+
+namespace seqan3
+{
+namespace test
+{
+#if defined(_WIN32)
+namespace
+{
+/*
+ * Helper function that implements mkdtemp for windows.
+ * Caveat:
+ *
+ * Race condition between processes exists.
+ * There exists the possibility that between generating
+ * an unique file name and creating the directory another
```

Comment:
```suggestion
 * a unique file name and creating the directory another
```
typo (this is nitpicking :smile:)

Reply:
No no, this is not nitpicking. It is a typo and it should be fixed as soon as possible :-)